src/go/k8s/pkg/resources/certmanager/kafka_api.go

```diff
@@ -67,22 +67,24 @@ func (r *PkiReconciler) prepareKafkaAPI(
 	nodeSecretRef := tlsListener.NodeSecretRef
 
 	if nodeSecretRef == nil {
-		// Redpanda cluster certificate for Kafka API - to be provided to each broker
-		cn := NewCommonName(r.pandaCluster.Name, RedpandaNodeCert)
-		certsKey := types.NamespacedName{Name: string(cn), Namespace: r.pandaCluster.Namespace}
+		certName := NewCertName(r.pandaCluster.Name, RedpandaNodeCert)
+		certsKey := types.NamespacedName{Name: string(certName), Namespace: r.pandaCluster.Namespace}
 		nodeIssuerRef := issuerRef
 		if externalIssuerRef != nil {
 			// if external issuer is provided, we will use it to generate node certificates
 			nodeIssuerRef = externalIssuerRef
 		}
 
-		dnsName := []string{r.internalFQDN}
 		externalListener := r.pandaCluster.ExternalListener()
-		if externalListener != nil && externalListener.External.Subdomain != "" {
-			dnsName = append(dnsName, externalListener.External.Subdomain)
+		internalListener := r.pandaCluster.InternalListener()
+		dnsNames := []string{}
+		if internalListener != nil && internalListener.TLS.Enabled {
+			dnsNames = append(dnsNames, r.internalFQDN)
 		}
```

Comment:
Why `dnsName := []string{r.internalFQDN}` was change to `dnsNames := []string{}` only in `kafka_api.go`? This theme is copied in every other file:
* schemaregistry_api.go
* admin_api.go
* pandaproxy_api.go

Reply:
Good catch.

I decided to refactor the code and added an `APIListener` interface to be able to reuse a single function for all API listeners.